app/timetables/tests/test_models.py

```diff
@@ -105,33 +106,70 @@ def test_duplicate_api_key_cannot_be_saved(self):
 
         self.assertRaises(ValidationError, self.another_timetable.save)
 
-    def test_current_cycle_day_greater_than_cycle_length_cannot_be_saved(self):
-        self.another_timetable.current_cycle_day = self.timetable.cycle_length + 1
+    def test_ref_cycle_day_greater_than_cycle_length_cannot_be_saved(self):
+        self.another_timetable.ref_cycle_day = self.timetable.cycle_length + 1
 
         self.assertRaises(ValidationError, self.another_timetable.save)
 
-    def test_cycle_length_and_current_cycle_day_of_zero_cant_be_saved(self):
+    def test_cycle_length_and_ref_cycle_day_of_zero_cant_be_saved(self):
         # test for cycle_length == 0
         self.another_timetable.cycle_length = 0
         self.assertRaises(ValidationError, self.another_timetable.save)
 
-        # test for current_cycle_day == 0
-        self.another_timetable.current_cycle_day = self.another_timetable.cycle_length
+        # test for ref_cycle_day == 0
+        self.another_timetable.ref_cycle_day = 0
         self.another_timetable.cycle_length = self.timetable.cycle_length
 
         self.assertRaises(ValidationError, self.another_timetable.save)
 
-    def test_cycle_length_and_current_cycle_day_of_negative_value_cant_be_saved(self):
+    def test_cycle_length_and_ref_cycle_day_of_negative_value_cant_be_saved(self):
         # test for cycle_length < 0
         self.another_timetable.cycle_length = -3
         self.assertRaises(ValidationError, self.another_timetable.save)
 
-        # test for current_cycle_day < 0
-        self.another_timetable.current_cycle_day = self.another_timetable.cycle_length
+        # test for ref_cycle_day < 0
+        self.another_timetable.ref_cycle_day = -3
         self.another_timetable.cycle_length = self.timetable.cycle_length
 
         self.assertRaises(ValidationError, self.another_timetable.save)
 
+    def test_calculate_cycle_day(self):
+        test_date = timezone.make_aware(timezone.datetime(2016, 11, 23, 12, 30, 0))
+        self.assertEqual(13, self.timetable.calculate_cycle_day(test_date))
+
+        test_date = timezone.make_aware(timezone.datetime(2016, 11, 24, 12, 30, 0))
+        self.assertEqual(14, self.timetable.calculate_cycle_day(test_date))
+
+        test_date = timezone.make_aware(timezone.datetime(2016, 11, 25, 12, 30, 0))
+        self.assertEqual(1, self.timetable.calculate_cycle_day(test_date))
+
+        test_date = timezone.make_aware(timezone.datetime(2016, 9, 25, 12, 30, 0))
+        self.assertEqual(None, self.timetable.calculate_cycle_day(test_date))
+
```

Comment:
@andela-kanyanwu: This is not really to test different cases of `cycle_day` rather it's to test that `calculate_cycle_day()` method accurately does its calculation specifically for dates that map to cycle days around the `cycle_length`; `cycle_day` should always wrap around the `cycle_length` by starting from 1 all over again after going through a full cycle (cycle_length). So values 13, 14 and 1 are considered here to fully test the `calculate_cycle_day()` method. The last case considered is that of a supplied date being earlier than the registered `ref_cycle_date`; in that case, `calculate_cycle_day()` method should resolve the supplied date to None.

The purpose of this test is to test the `calculate_cycle_day()` method as a whole. From the point of view of the class it belongs to, it's already modular. I feel going to the extent of dedicating a test method for each pair of precondition and postcondition is not necessary because whichever pair that causes the test to fail would always be known at runtime.